tests/Progress/ProgressHashesTest.php

```diff
@@ -0,0 +1,57 @@
+<?php
+
+namespace BristolSU\Support\Tests\Progress;
+
+use BristolSU\Support\Progress\ProgressHashes;
+use BristolSU\Support\Tests\TestCase;
+use Illuminate\Foundation\Testing\WithFaker;
+
+class ProgressHashesTest extends TestCase
+{
+    use WithFaker;
+
+    protected $Model;
+    protected $Table;
+
+    public function setUp(): void
+    {
+        parent::setUp();
+
+        $this->Model = (new ProgressHashes);
+        $this->Table = $this->Model->getTable();
+    }
+
+    public function getModelData()
+    {
+        return [
+            'item_key' => $this->faker('en_UK')->slug,
+            'hash' => $this->faker('en_UK')->uuid
+        ];
+    }
+
+    /** @test */
+    public function a_model_can_be_saved_and_retrieved_by_the_hash(){
+        $data = $this->getModelData();
+
+        $this->Model->create($data);
+
+        $this->assertDatabaseHas($this->Table, $data);
+
+        $dbValue = $this->Model->getByHash($data['hash'])->first(['item_key', 'hash'])->toArray();
+
+        $this->assertEquals($data, $dbValue);
```

Comment:
You should be able to call `$this->assertModelEquals($this->model, $dbValue);` where $dbValue is the actual model NOT an array. This uses the `$model->is()` function behind the scenes, so will just handle cases where extra columns are added to the model, attributes are made hidden etc

Reply:
This doesn't work as the $this->Model is an empty instance of the Model, so we'd have to request the data the same way as the $dbValue.